newsService/src/main/java/com/kovan/app/service/NewsService.java

```diff
@@ -63,25 +72,51 @@ public NewsDto getTopHeadlines() {
 
         } while (pageSize <= newsDto.getTotalResults());
 
-        LocalDate dateOnly = LocalDate.parse(
-                articles.getFirst().getPublishedAt(),
-                DateTimeFormatter.ISO_DATE_TIME
-        );
+        List<NewsDto.Article> newArticles = new ArrayList<>();
+        List<NewsDto.Article> latestArticles = new ArrayList<>();
 
-        if (!isNewsAlreadyInDb(dateOnly.toString())) {
-            NewsDto finalDto = NewsDto.builder()
-                    .articles(articles)
-                    .totalResults(articles.size())
-                    .publishedAt(dateOnly.toString())
-                    .build();
+        boolean isYesterdayInDb = isNewsAlreadyInDb(yesterday.toString());
+        boolean isTodayInDb = isNewsAlreadyInDb(yesterday.plusDays(1).toString());
```

Comment:
why yesterday.plusDays(1). you know it is today right directly you can use today .